src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -59,6 +61,72 @@ to_soa_layout(const std::vector<ingress_egress_state<T>>& v) {
 
 namespace {
 
+const auto direction_label = ss::metrics::label("direction");
+const auto label_ingress = direction_label("ingress");
+const auto label_egress = direction_label("egress");
+
+template<class T>
+void add_ingress_egress_gauges(
+  std::vector<ss::metrics::metric_definition>& metric_defs,
+  ss::metrics::metric_name_type name,
+  T&& value_getter,
+  ss::metrics::description desc,
+  const std::vector<ss::metrics::label>& aggregate_labels) {
+    metric_defs.emplace_back(ss::metrics::make_gauge(
+                               name,
+                               [value_getter] { return value_getter().in; },
+                               desc,
+                               {label_ingress})
+                               .aggregate(aggregate_labels));
+    metric_defs.emplace_back(ss::metrics::make_gauge(
+                               name,
+                               [value_getter] { return value_getter().eg; },
+                               desc,
+                               {label_egress})
+                               .aggregate(aggregate_labels));
+}
+
+} // namespace
+
+void snc_quotas_probe::setup_metrics() {
+    namespace sm = ss::metrics;
+
+    if (config::shard_local_cfg().disable_metrics()) {
+        return;
+    }
+
+    const auto aggregate_labels = config::shard_local_cfg().aggregate_metrics()
+                                    ? std::vector<sm::label>{sm::shard_label}
+                                    : std::vector<sm::label>{};
+    std::vector<ss::metrics::metric_definition> metric_defs;
+
+    if (ss::this_shard_id() == quota_balancer_shard) {
+        metric_defs.emplace_back(sm::make_counter(
+          "balancer_runs",
+          [this] { return _balancer_runs; },
+          sm::description(
+            "{}: Number of times throughput quota balancer has been run")));
+    }
+    add_ingress_egress_gauges(
+      metric_defs,
+      "quota_effective",
+      [this] { return _qm.get_quota(); },
+      sm::description("{}: Currently effective quota, in bytes/s"),
+      aggregate_labels);
+    add_ingress_egress_gauges(
+      metric_defs,
+      "throughput",
+      [this] { return _qm.get_throughput(); },
+      sm::description("{}: Throughput measurement at the time of last "
+                      "balancer run, in bytes/s"),
+      aggregate_labels);
+
+    _metrics.add_group(
+      prometheus_sanitize::metrics_name("kafka:quotas_snc"), metric_defs);
```

Comment:
what is `snc`? maybe something more descriptive for metrics?

Reply:
Good point, but not sure what. I was trying to express the classificatin of the kafka API related quotas which can be
- client related (per-client_id or per-client_group)
- global to shard, node, and cluster (hence SNC)

Calling the latter 'global' is not descriptive (global as opposed to what?...), calling them 'shard' or 'shard-wide' means downplaying node and cluster scopes which leads to confusion... Still a disstinction from client quotas is required.

Any ideas are welcome.

Reply:
perhaps there is only one meaningful quota active at once we can use a single name?

Reply:
nothing stops a customer from using both quotas actually. But since other quotas don't report metrics yet, I can use the right of the first and occupy the plain 'quota' identifier.

Reply:
yeh it's a minor point. other folks may have more helpful input here. we can raise it in slack later